boards/arm64/intel_socfpga_agilex_socdk/intel_socfpga_agilex_socdk_defconfig

```diff
@@ -24,3 +24,9 @@ CONFIG_CLOCK_CONTROL=y
 
 # For Misc Register Map
 CONFIG_SYSCON=y
+
+# Misc
+CONFIG_HEAP_MEM_POOL_SIZE=16384
+
+# Enable QSPI Flash
+CONFIG_FLASH=y
```

Comment:
please justify why these are minimal settings required for all/majority of applications.

Reply:
In near future, we would upstream a full feature of a remote system upgrade, which require flash to be enabled per our implementation, enabling flash at application (eg: at flash_shell only) the remote system upgrade may not work.

Reply:
I don't understand, if the prj.conf is at Flash sample. How the remote system upgrade being set to use the flash?

Reply:
The only way i can think of is, when i enable the remote system upgrade feature, the CONFIG FLASH option will be added in the prj.conf at the remote system upgrade samples. Hence moving out the Flash config from board.

Reply:
@ngboonkhai  the defconfig of a board shall not be seen as the application configuration layer. The board configuration should be implemented with the minimal set of configuration options that can be then extended in the application. Reason for this is to allow different applications and use cases to derive from the same board definition wihout having to deal with conflicts and disabling optional features enabled by the board on the application level.

Application configuration should be ideally additive and should not be about removing features, so if there are situations where CONFIG_FLASH would not be needed, it should not be part of the defconfig, however, if this board *will* not boot or operate without this option enabled, then it can be part of the board configuration.

Reply:
Hi Anas, that case, can I say, if I go to search for the samples, if any one of the samples can operate without the CONFIG_FLASH option, then the CONFIG_FLASH should not be added at the board?